packages/core/src/delegatedTaggedState.spec.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable jest/expect-expect */
+import { AssertEqual } from "@tagged-state/testutils";
+import { DelegatedFlatStateVariant } from "./delegatedTaggedState";
+import { StateProps, StateVariant } from "./taggedState";
+
```

Comment:
We should have a test that ensures type equality between the flat state and the corresponding tagged state.

Similar to `correctly derives StateProps`, but without involving `StateProps`.

Reply:
https://github.com/MOZGIII/tagged-state/pull/1/commits/d8154e1df95ee0cee6d3be9dc4d32727a4c870e5